pkg/vault/test/foundry/BatchRouterERC4626Pool.t.sol

```diff
@@ -933,4 +863,39 @@ contract BatchRouterERC4626PoolTest is BaseERC4626BufferTest {
         IERC20(address(newPool)).approve(address(batchRouter), type(uint256).max);
         vm.stopPrank();
     }
+
+    struct BufferBalances {
+        uint256 underlying;
+        uint256 wrapped;
+    }
+
+    struct TestBalances {
+        BaseVaultTest.Balances balances;
+        BufferBalances waUSDCBuffer;
+        BufferBalances waDAIBuffer;
+        uint256 daiIdx;
+        uint256 usdcIdx;
+        uint256 waDaiIdx;
+        uint256 waUsdcIdx;
+    }
+
+    function _getTestBalances(address sender) private view returns (TestBalances memory testBalances) {
+        IERC20[] memory tokenArray = [address(dai), address(usdc), address(waDAI), address(waUSDC)]
+            .toMemoryArray()
+            .asIERC20();
+        testBalances.balances = getBalances(sender, tokenArray);
+
+        (uint256 waDAIBufferBalanceUnderlying, uint256 waDAIBufferBalanceWrapped) = vault.getBufferBalance(waDAI);
+        testBalances.waDAIBuffer.underlying = waDAIBufferBalanceUnderlying;
+        testBalances.waDAIBuffer.wrapped = waDAIBufferBalanceWrapped;
+
+        (uint256 waUSDCBufferBalanceUnderlying, uint256 waUSDCBufferBalanceWrapped) = vault.getBufferBalance(waUSDC);
+        testBalances.waUSDCBuffer.underlying = waUSDCBufferBalanceUnderlying;
+        testBalances.waUSDCBuffer.wrapped = waUSDCBufferBalanceWrapped;
+
+        testBalances.daiIdx = 0;
+        testBalances.usdcIdx = 1;
+        testBalances.waDaiIdx = 2;
+        testBalances.waUsdcIdx = 3;
```

Comment:
The index of each token is defined by the order of tokenArray. Since the tokenArray is defined in the same function, we can for sure know each index and hardcode it. I could use sort methods too, but looked much more complex than it should be. I added a comment in the code explaining this.